suite-native/module-add-accounts/src/screens/SelectAccountTypeScreen.tsx

```diff
@@ -0,0 +1,149 @@
+const gradientStyle = prepareNativeStyle(_ => ({
+    height: GRADIENT_HEIGHT,
+}));
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.medium,
+    backgroundColor: utils.colors.backgroundSurfaceElevation0,
+}));
+
+const aboutStyle = prepareNativeStyle((utils, { bottomInset }: { bottomInset: number }) => ({
+    paddingTop: utils.spacings.extraLarge,
+    paddingBottom: 2 * GRADIENT_HEIGHT + bottomInset,
+    width: '100%',
+    paddingHorizontal: utils.spacings.small,
+    gap: 12,
+}));
+
+export const SelectAccountTypeScreen = ({
+    route,
+}: StackProps<AddCoinAccountStackParamList, AddCoinAccountStackRoutes.SelectAccountType>) => {
+    const { defaultType, network } = route.params;
+    const { translate } = useTranslate();
+    const openLink = useOpenLink();
+    const insets = useSafeAreaInsets();
+    const { applyStyle, utils } = useNativeStyles();
+    const navigation =
+        useNavigation<
+            StackNavigationProps<
+                AddCoinAccountStackParamList,
+                AddCoinAccountStackRoutes.SelectAccountType
+            >
+        >();
+
+    const { getAvailableAccountTypesForNetwork, addCoinAccount, isAddingCoinAccount } =
+        useAddCoinAccount();
+    const [selectedAccountType, setSelectedAccountType] = useState<AccountType>(defaultType);
+
+    const onPressSelectedAccountType = (type: AccountType) => setSelectedAccountType(type);
+
+    const types: AccountType[] = getAvailableAccountTypesForNetwork({ network });
+    const { titleKey: accountTypeKey } = getIntlKeysForAccountType(selectedAccountType);
+
+    const handleClose = () => navigation.goBack();
+
+    const handleMoreTap = () => openLink(ACCOUNT_TYPES_URL);
+
+    const handleConfirmTap = () => {
+        if (!isAddingCoinAccount) {
+            addCoinAccount({ network, accountType: selectedAccountType });
+        }
+    };
+
+    return (
+        <>
+            <Screen
+                screenHeader={
+                    <ScreenSubHeader
+                        content={translate('moduleAddAccounts.selectAccountTypeScreen.title', {
+                            symbol: _ => network.symbol.toUpperCase(),
+                        })}
+                        leftIcon={
+                            <IconButton
+                                iconName="close"
+                                onPress={handleClose}
+                                colorScheme="tertiaryElevation0"
+                                size="medium"
+                            />
+                        }
+                    />
+                }
+            >
+                <VStack spacing="large" style={applyStyle(itemsStyle)}>
+                    {types.map(item => (
+                        <SelectableAccountTypeItem
+                            key={`type-${item}`}
+                            isSelected={selectedAccountType === item}
+                            isDefault={defaultType === item}
```

Comment:
Afaik the rule is not to send anything that is not to send whole objects when you can pass only some simple information. The component does not have to know how to decide whether it is default or not.

Reply:
But you are already passing the whole object on line 119 😄

Reply:
Good spot. I should probably send only the texts, but this will complicate the code here much more. So I am unsure what approach is the best, but to me it feels, that the current approach is slightly better.